appservice/src/utils/workspace.ts

```diff
@@ -5,68 +5,55 @@
 
 import * as path from 'path';
 import * as vscode from 'vscode';
-import { IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { IActionContext, IAzureQuickPickItem } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 
-export async function selectWorkspaceFolder(placeHolder: string, getSubPath?: (f: vscode.WorkspaceFolder) => string | undefined | Promise<string | undefined>): Promise<string> {
+export async function selectWorkspaceFolder(context: IActionContext, placeHolder: string): Promise<string> {
     return await selectWorkspaceItem(
+        context,
         placeHolder,
         {
             canSelectFiles: false,
             canSelectFolders: true,
             canSelectMany: false,
             defaultUri: vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0 ? vscode.workspace.workspaceFolders[0].uri : undefined,
             openLabel: localize('select', 'Select')
-        },
-        getSubPath);
+        });
 }
 
-export async function selectWorkspaceFile(placeHolder: string, getSubPath?: (f: vscode.WorkspaceFolder) => string | undefined | Promise<string | undefined>, fileExtensions?: string[]): Promise<string> {
```

Comment:
Doesn't this affect the Java deploy user scenario? It wouldn't have the defaultUri anymore or automatically detect the subpath and automatically select the effectiveDeployFsPath if only one folder is opened.

Reply:
talked offline